pkg/geo/wkt/wkt.y

```diff
@@ -41,6 +41,16 @@ type geomPair struct {
 	ends       []int
 }
 
+func appendGeomPairs(p1 geomPair, p2 geomPair) geomPair {
+	if len(p1.ends) > 0 {
+		p1LastEnd := p1.ends[len(p1.ends)-1]
+		for i, _ := range p2.ends {
```

Comment:
i think the way this is written with the yacc makes this n^2.
is it possible to do `four_coords_point_list_allowing_empty_point ',' four_coords_point_list_allowing_empty_points` such that it is not? alternatively, can we structure this such that we only finalise `ends` using one pass?

Reply:
(i could be wrong about this being n^2 though, depends in the order in which the tokens are resolved)

Reply:
I think since the grammar rules are left recursive, `len(p2.ends)` will always be 1 so the `append` should happen in amortized O(1), making it O(n) expected time overall?

Reply:
>  alternatively, can we structure this such that we only finalise ends using one pass?

I think doing this would still require appends so the time complexity would be the same?

Reply:
yeah ok! thanks for the research and entertaining my random thoughts

Reply:
No problem! Thanks for the review!